src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -405,6 +420,15 @@ public async Task SaveChangesAsync(IList<UnspentOutputs> unspentOutputs, IEnumer
 
                     cacheItem.IsDirty = true;
 
+                    if (this.rewindDataIndexStore != null)
+                    {
+                        for (int i = 0; i < unspent.Outputs.Length; i++)
+                        {
+                            string key = $"{unspent.TransactionId}-{i}";
+                            indexItems[key] = checked((int)unspent.Height);
```

Comment:
just use `uint`

Reply:
yes declare 
`var indexItems = new Dictionary<string, int>();`
as
`var indexItems = new Dictionary<string, uint>();`
and you can just add unspent.Height without casting it

Reply:
of course this affect all the other methods that are expecting a Dictionary<string, int> because now would be Dictionary<string, uint>
Nonetheless I'd use the same type for represents the same things along our codebase and actually we have a mix of int/uint, @dangershony what should we use ?